src/javascript/carouselBox.js

```diff
@@ -0,0 +1,124 @@
+export default function CarouselBox({ target, pagingTarget, prefix, list }) {
+  this.target = target;
+  this.pagingTarget = pagingTarget;
+  this.prefix = prefix;
+  this.list = list;
+  this.leftArrow = null;
+  this.rightArrow = null;
+  this.pagingInterval = null;
+}
+
+CarouselBox.prototype = {
+  init,
+  renderCarouselBox,
+  getCarouselBoxBlock,
+  setTargetInitialStyle,
+  renderArrow,
+  createArrowEvent,
+  moveCarouselBox,
+  shuffleCarouselBoxItem
+}
+
+CarouselBox.constructor = CarouselBox;
+
+function init() {
+  this.renderCarouselBox();
+  this.setTargetInitialStyle();
+  this.renderArrow();
+  this.createArrowEvent();
+}
+
+function renderCarouselBox() {
+  const boxList = this.list.reduce((acc, item) => acc += this.getCarouselBoxBlock(item), '');
+  this.target.insertAdjacentHTML('beforeend', boxList);
+}
+
+function getCarouselBoxBlock(item) {
+  return `
+    <a class="box__item carousel__item" href="">
+      <img class="box__item__img" src="${this.prefix}${item.src}">
+      <span class="box__item__title">${item.title}</span>
+      <span class="box__item__content">${item.content}</span>
+      <span class="box__item__tag">${item.tag}</span>
+    </a>
+  `;
+}
+
+function setTargetInitialStyle() {
+  const { target, list } = this;
+  target.style.width = `${list.length * 20}%`;
+  setTimeout(() => {
+    const childWidth = target.firstElementChild.getBoundingClientRect().width;
+    target.dataset.width = childWidth;
+    target.dataset.default_width = childWidth * -parseInt(list.length / 2);
+    target.style.transform = `translateX(${target.dataset.default_width}px)`;
+  }, 300);
+}
+
+function renderArrow() {
+  const arrowPrefixClassName = 'box-paging__arrow';
+  const arrowWrap = document.createElement('div');
+  arrowWrap.classList.add(arrowPrefixClassName);
+
+  const leftArrow = document.createElement('div');
+  leftArrow.classList.add(`${arrowPrefixClassName}__left`);
+  leftArrow.dataset.arrow = 'prev';
+
+  const rightArrow = document.createElement('div');
+  rightArrow.classList.add(`${arrowPrefixClassName}__right`);
+  rightArrow.dataset.arrow = 'next';
+
+  arrowWrap.append(leftArrow);
+  arrowWrap.append(rightArrow);
+  this.pagingTarget.append(arrowWrap);
+  this.leftArrow = leftArrow;
+  this.rightArrow = rightArrow;
+}
+
+function createArrowEvent() {
+  this.pagingTarget.addEventListener('mousedown', ({ target }) => {
+    this.pagingInterval = setInterval(() => {
+      this.moveCarouselBox(target.dataset.arrow, 2);
+      target.dataset.moved = '1';
+    }, 2000);
+  });
+  this.pagingTarget.addEventListener('mouseup', ({ target }) => {
+    if(!target.dataset.moved) {
+      this.moveCarouselBox(target.dataset.arrow);
+    }
+    delete target.dataset.moved;
+    clearInterval(this.pagingInterval);
+  });
+}
+
+function moveCarouselBox(type, count = 1) {
+  if(!type) return;
+  const { target } = this;
+  let move = +target.dataset.width * count;
+  if(type === 'next') {
+    move *= -1;
+  }
```

Comment:
형변환을 명시적으로 하는 것이 좋아보여요. 그리고 변수 선언을 정말 불가피한 경우가 아니면 const로 하는 것이 좋다고 생각해요. let으로 선언하게 되면 해당 변수가 유효한 스코프 내에서 재할당 하는 코드가 있는지 끝까지 확인해야 합니다.
```suggestion
  const direction = type === 'next' ? -1 : 1;
  const move = Number(target.dataset.width) * count * direction;
```

Reply:
> 형변환을 명시적으로 하는 것이 좋아보여요. 그리고 변수 선언을 정말 불가피한 경우가 아니면 const로 하는 것이 좋다고 생각해요. let으로 선언하게 되면 해당 변수가 유효한 스코프 내에서 재할당 하는 코드가 있는지 끝까지 확인해야 합니다.

형변환을 명시적으로... 메모하겠습니다!☑️ 
let을 최대한 안쓰는 방향을 좀 더 생각해봐야겠네요!
제안해주신 코드를 보고 나니 굳이 let으로 할 필요가 없었겠구나 생각됩니다 ::)